sources/packages/backend/libs/integrations/src/cas/cas.service.ts

```diff
@@ -267,6 +246,28 @@ export class CASService {
     return supplierAddress;
   }
 
+  /**
+   * Handles bad request errors from CAS API.
+   * @param error The error object caught during the request.
+   * @param defaultMessage The default message to throw for non-specific errors.
+   */
+  private handleBadRequestError(error: unknown, defaultMessage: string): never {
```

Comment:
In some of the upcoming PRs lets add a unit test to validate the `CustomNamedError`.